# **State the Hypotheses**

### **Null Hypothesis (H₀):**

* The two variables are independent.  
* The distribution of one variable is the same across the categories of the other variable.
*   The observed frequencies match the expected frequencies under the assumption of independence.


### **Alternative Hypothesis (H₁):**


*   The two variables are related (associated).
*   The distribution of one variable differs across the categories of the other variable.
*   The observed frequencies differ significantly from the expected frequencies.


# **Compute the Chi-Square Statistic**

chi_square_statistic = np.sum((observed - expected)**2 / expected)

  observed: An array of observed frequencies for each category.

  expected: An array of expected frequencies for each category.

  The np.sum function sums up the individual values across the array.

  Divide the squared difference by the expected frequency for each category.
  
  Sum the results for all categories.

In [1]:
import numpy as np
observed = np.array([
    [50, 70],
    [80, 100],
    [60, 90],
    [30, 50],
    [20, 50]
])
print(observed)

[[ 50  70]
 [ 80 100]
 [ 60  90]
 [ 30  50]
 [ 20  50]]


In [5]:
import pandas as pd
df=pd.DataFrame({'satisfaction':['Very Satisfied','Satisfied','Neutral','Unsatisfied','Very Unsatisfied'],'Smart Thermostat':[50,80,60,30,20],'Smart Light':[70,100,90,50,50]})
df

satisfaction  Smart Thermostat  Smart Light
0    Very Satisfied                50           70
1         Satisfied                80          100
2           Neutral                60           90
3       Unsatisfied                30           50
4  Very Unsatisfied                20           50

In [6]:
row = observed.sum(axis=1)
column = observed.sum(axis=0)
total = observed.sum()
print("observation total :",total)

observation total : 600


In [7]:
expected = np.outer(row,column)/total
chi_square_statistic = ((observed-expected)**2/expected).sum()
print("chi_square statistic :", (chi_square_statistic.round(3)))

chi_square statistic : 5.638


# **Determine the Critical Value**

In [9]:
import scipy.stats as stats
critical_value = stats.chi2.ppf(q=0.95,df=4)
print("Critical value :",critical_value.round(3))

Critical value : 9.488


Find Critical Value: The critical value for the chi-square statistic at alpha=0.05 and df = 4 can be found using a chi-square distribution

Find Degrees of Freedom (df): Using the formula

df = (number_rows-1)(number_columns-1), given rows are 5 and columns is 2

If your calculated chi-square statistic exceeds this critical value, you can reject the null hypothesis. Otherwise, you fail to reject the null hypothesis.

# **Make a Decision**

If the Chi-Square statistic is less than critical value, we fail to reject the null hypothesis.

5.638<9.488,we fail to reject the null hypothesis

The Chi-Square statistic (5.545) is less than the critical value (9.488) at the 0.050 significance level.

we fail to reject the null hypothesis. This means that there is not enough evidence to suggest that the observed frequencies differ significantly from the expected frequencies.